lib/xendit_api/middleware/handle_response_exception.rb

```diff
@@ -105,6 +105,8 @@ def validate_response(response)
         else
           raise XenditApi::Errors::UnknownError.new(error_message, json_response)
         end
+      rescue JSON::ParserError
+        raise XenditApi::Errors::ServerError
```

Comment:
Hmn, please don't handle it like this ya. JSON::ParserError is the exception that we get when we fail to parse the response, not explained the Xendit server error. I prefer to handle like this:
```rb
def call(env)
  @app.call(env).on_complete do |response|
    raise Xendit::Api::Errors::ServerError, 'Xendit server error' if response.status.to_s.start_with?("5")

    validate_response(response.body)
  end
end
```

wdyt, @rubyvu ?

Reply:
ok, let me modify it. Thanks

Reply:
response.status.to_s.start_with?("5") I think it will have problem if response is null is there a case response is null? because I see on validate_response method the first line is check for response.nil?

Reply:
Yes, there is a case if the response is `null` -- which is the Xendit server error (500 status code). 

If that happened, we need to raise the `ServerError` exception Vu. And, let's say if the response is null but the status code is `201` that does not cause any problem right? since we only raise the `ServerError` after we make sure the response status was 5xx. 

And I think you can read some references on the Xendit errors from this link Vu: https://developers.xendit.co/api-reference/#errors.

Reply:
I mean, if response is null. response.status will cause NoMethodError, isn't it?